src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncProgressResolver.cs

```diff
@@ -117,6 +124,25 @@ public long FindBestFullState()
                 }
             }
 
+            if (bestFullState == 0)
+            {
+                if (_bestStateCounter != -1)
+                {
+                    return _bestStateCounter;
+                }
+                byte[] bestStatePersistedCounter = _stateDb.Get(LatestBlockWithFullState);
```

Comment:
use metadata db if possible. I think we will try to keep state db pure in the future to better support iterators on it.